recipe_Menager.py

```diff
@@ -0,0 +1,42 @@
+class RecipeManager:
+    def __init__(self):
+        self.recipes = []
+
+    def add_recipe(self, recipe):
+        self.recipes.append(recipe)
+
+    def update_recipe(self, recipe_index, new_recipe):
+        self.recipes[recipe_index] = new_recipe
+      #on this point I am confused, because I can see 2 meanings:
+      # 1. change older recipe for new one and I had choosen this version
+      # 2. change the integrent in recipe
+
+    def delete_recipe(self, recipe_index):
+        self.recipes.pop(recipe_index)
```

Comment:
An index is probably not a bullet proof way to reference a recipe, its better if the recipe is "found" by a unique form - or in a simple form the "name" in the list and then that reference used to remove rather than pop.

Reply:
I just was thinking that not everyone will want to tape the name of the recipe, just will be easier to add them like number list like:
1. Spaghetti
2. Scrabble egg
and just removing from this format. But if that need to be fixing I will do it on night time.

Reply:
If we arrange the search correctly we could either use wild card options or partial matches to bring up a truncated list that can be selected from.

Reply:
So one of the problems with using indexes is best described here https://replit.com/@LachlanMackenz3/PopvsRemove#main.py - because indexes change as lists change unless you can guarantee that the index is correct after something else is removed from a list it can cause no end of problems and this is a side effect of using a reference to an object that has no bearing on what the actually is (simply its position within a data structure like a list).

Reply:
In terms of searching for a recipe - that essentially needs a new function, but to simply select a recipe from a graphical interface it should not be difficult to create the numbered ordering in the GUI based upon a list of Recipes, when the user selects a recipe the graphical interface "marries up" the selection with a particular recipe class instance and its this that can be used during the delete.